ospfd/ospf_flood.c

```diff
@@ -222,12 +225,51 @@ static void ospf_process_self_originated_lsa(struct ospf *ospf,
 			ospf_translated_nssa_refresh(ospf, NULL, new);
 			return;
 		}
+
+		al = (struct as_external_lsa *)new->data;
+		p.family = AF_INET;
+		p.prefixlen = ip_masklen(al->mask);
+		p.prefix = new->data->id;
+
 		ei = ospf_external_info_check(ospf, new);
-		if (ei)
+		if (ei) {
+			if (ospf_external_aggr_match(ospf, &ei->p)) {
+				if (IS_DEBUG_OSPF(lsa, EXTNL_LSA_AGGR))
+					zlog_debug(
+						"%s, Matching external aggregate route found for %s, so don't refresh it.",
+						__func__,
+						inet_ntoa(ei->p.prefix));
```

Comment:
use `%pI4` to print ipv4 addresses rather than `inet_ntoa`

Reply:
done